cmd/api/main_test.go

```diff
@@ -0,0 +1,233 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/alexdunne/gs-onboarding/hn"
+)
+
+type StubItemStore struct {
+	items            map[string]hn.Item
+	getAllCalled     int
+	getStoriesCalled int
+	getJobsCalled    int
+}
+
+func (s *StubItemStore) GetAll() hn.Items {
+	s.getAllCalled++
+
+	items := hn.Items{}
+	for _, v := range s.items {
+		items = append(items, v)
+	}
+
+	return items
+}
+
+func (s *StubItemStore) GetStories() hn.Items {
+	s.getStoriesCalled++
+
+	items := hn.Items{}
```

Comment:
This is all good, but you might want to look at mocks when you get more services involved, heres an example from realm migration service https://github.com/gymshark/realm-migration-service/blob/40c096f7018e289dd34b74201f9490c859c328f3/internal/mongo/collection_test.go#L21

You can set up the mock, add expected calls and returns and then assert on them

Reply:
I looked into using additional testing tools but it seems the vocal go community on the internet are against it :D I'll take a look 👍